app/export.go

```diff
@@ -118,26 +177,65 @@ func (a *App) BulkExport(ctx request.CTX, writer io.Writer, outPath string, opts
 	if err != nil {
 		return err
 	}
-
+	exportFilename := a.GetID() + "_export.log"
+	var missingAttachments []LogFormat
 	if opts.IncludeAttachments {
-		ctx.Logger().Info("Bulk export: exporting file attachments")
+		mlog.Info("Bulk export: exporting file attachments")
 		for _, attachment := range attachments {
-			if err := a.exportFile(outPath, *attachment.Path, zipWr); err != nil {
+			if err := a.ExportAttachments(attachment, exportFilename, &missingAttachments, outPath, zipWr); err != nil {
 				return err
 			}
 		}
 		for _, attachment := range directAttachments {
-			if err := a.exportFile(outPath, *attachment.Path, zipWr); err != nil {
+			if err := a.ExportAttachments(attachment, exportFilename, &missingAttachments, outPath, zipWr); err != nil {
 				return err
 			}
 		}
-		for _, emojiPath := range emojiPaths {
-			if err := a.exportFile(outPath, emojiPath, zipWr); err != nil {
+		for _, emoji := range emojiList {
+			isExist, err := a.FileExists(*emoji.Path)
+			if err != nil {
+				return err
+			}
+			if !isExist {
+				createdAtTime, err := strconv.ParseInt(strconv.FormatInt(emoji.CreateAt, 10), 10, 64)
```

Comment:
can u explain why did you do this rather than just using emoji.CreateAt

Reply:
attachment.CreateAt is in integer form so i have to convert it into time.Time

Reply:
strconv.FormatInt converts a int to string and strconv.ParseInt converts a string to int why are you converting them 2 times if you obly require an integer ?

Reply:
Done